forest/cli/src/cli/info_cmd.rs

```diff
@@ -0,0 +1,402 @@
+// Copyright 2019-2023 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use chrono::{DateTime, Utc};
+use clap::Subcommand;
+use colored::*;
+use forest_blocks::tipset_json::TipsetJson;
+use forest_cli_shared::cli::CliOpts;
+use forest_rpc_client::{
+    chain_get_name, chain_head, node_ops::node_status, start_time, wallet_balance,
+    wallet_default_address,
+};
+use forest_shim::econ::TokenAmount;
+
+use fvm_shared::clock::EPOCH_DURATION_SECONDS;
+use fvm_shared::{clock::ChainEpoch, BLOCKS_PER_EPOCH};
+use humantime::format_duration;
+use num::BigInt;
+use std::time::{Duration, SystemTime, UNIX_EPOCH};
+
+use super::Config;
+use crate::cli::handle_rpc_err;
+use crate::humantoken::TokenAmountPretty;
+
+#[derive(Debug, Subcommand)]
+pub enum InfoCommand {
+    Show,
+}
+
+#[derive(Debug)]
+pub struct NodeStatusInfo {
+    /// How far behind the node is with respect to syncing to head
+    pub lag: Duration,
+    /// Chain health is the percentage denoting how close we are to having
+    /// an average of 5 blocks per tipset in the last couple of
+    /// hours. The number of blocks per tipset is non-deterministic
+    /// but averaging at 5 is considered healthy.
+    pub health: f64,
+    /// epoch the node is currently at
+    pub epoch: ChainEpoch,
+    /// Base fee is the set price per unit of gas (measured in attoFIL/gas unit) to be burned (sent to an unrecoverable address) for every message execution
+    pub base_fee: TokenAmount,
+    pub sync_status: SyncStatus,
+    /// Start time of the node
+    pub start_time: Option<DateTime<Utc>>,
+    pub network: Option<String>,
+    pub default_wallet_address: Option<String>,
+    pub default_wallet_address_balance: Option<String>,
+    use_color: bool,
+}
+
+impl NodeStatusInfo {
+    fn chain_status(&self) -> ColoredString {
+        let base_fee_fmt = self.base_fee.pretty();
+        let behind = format!("{}", humantime::format_duration(self.lag));
+        let chain_status = format!(
+            "[sync: {}! ({} behind)] [basefee: {base_fee_fmt}] [epoch: {}]",
+            self.sync_status, behind, self.epoch
+        )
+        .blue();
+
+        if !self.use_color {
+            chain_status.clear()
+        } else {
+            chain_status
+        }
```

Comment:
You repeat this snippet of code six times. Let's refactor such that we only clear the color once.

Reply:
Fixed